### DISCUSSION TITLE
CacheStore - ForgetMany

### DISCUSSION THREAD

**Discussion by Mat-Moo** [laravel/framework#53622] [Ideas]
We have Put, PutMany, get, many, but there is no way to forget more than one item at a time.
Use case - We have a job that pulls in data from a table, these are matched against redis keys, and then merged with a second array to process. For each one of these rows (can be up to 10k+)  we have to update the db row and forget the cache key. Updating 10k rows in a db is outside the loop with a bulk update, but we have to make 10k+ calls to forget redis keys in the loop... not ideal

I'm not sure about other cache system, but redis DEL command already can take mulitple keys

**Comment by henzeb** [comment]
It will probably end up being a foreach loop anyway. PutMany works that way.

**Reply by Mat-Moo** [reply]
Redis del takes multiple parameters...

**Reply by henzeb** [reply]
Redis does, the way laravel has set up the driver does not....

**Reply by Mat-Moo** [reply]
Hence why I'm suggesting a new method to follow along the lines of the other Many versions, if you can put many why not forgetMany?

**Reply by henzeb** [reply]
Well, I finally had some peace and quiet and I saw we already have a deleteMultiple.

https://github.com/laravel/framework/blob/abc1faa60887cb54b4050277e07f0e4f25244a5f/src/Illuminate/Cache/Repository.php#L557

As you can see here, it just loops through the array of keys and then calls `forget`.

as for many/getMultiple:

https://github.com/laravel/framework/blob/abc1faa60887cb54b4050277e07f0e4f25244a5f/src/Illuminate/Cache/Repository.php#L140

That also loops through the keys.

changing that also requires changing all the drivers. It will probably be a change for 12.x.
